scrapy/utils/request.py

```diff
@@ -68,6 +70,67 @@ def request_fingerprint(request, include_headers=None, keep_fragments=False):
     return cache[cache_key]
 
 
+_fingerprint_cache = weakref.WeakKeyDictionary()
+
+
+def fingerprint(request, include_headers=None, keep_fragments=False) -> bytes:
```

Comment:
A chance for a clean-up :)

```suggestion
def fingerprint(request, *, include_headers=None, keep_fragments=False) -> bytes:
```